### ISSUE TITLE
Request Entity Too large / Events with loaded data > 100kbs cannot be processed

### ISSUE THREAD

**Issue by yasserlens** [vendure-ecommerce/vendure#2564]
**Describe the bug**
tldr; requests with POST body exceeding 100KBs result in a "Request Entity Too Large" error. Middleware solutions [like this one](https://docs.vendure.io/reference/typescript-api/common/middleware/#increasing-the-maximum-request-body-size-limit) do not work on Vendure v2.1.4 for 2 reasons: We get a "POST body missing, invalid Content-Type, or JSON object has no keys." error, and Stripe expects raw body requests (not JSON).

Long version:
We want to use the Email plugin to send transactional emails such as order confirmation messages. However, we can't get such requests because they're too large. Here's what happens:
- Order state changes to PaymentSettled
- Through EmailEventListener, we listen to such events, and load order data to them before passing them to Email handlers which consume the data to generate the email templates.
- Our Vendure instance is on Google Cloud Run and we use Google Cloud Tasks to handle Vendure Worker processes.
- Our Order objects are large (>100kbs) due to the attached CustomEntities. Loading this data causes the issue described in the title.
- The [solution described here](https://docs.vendure.io/reference/typescript-api/common/middleware/#increasing-the-maximum-request-body-size-limit) won't work because it results in errors in Admin UI and Stripe.

**To Reproduce**
See above. To replicate, create a EmailEventListener, attach large objects in the loadData() method to exceed the 100kbs.

**Expected behavior**
Ideally, we should be able to increase the 100kb limit to receive larger requests without getting errors in Stripe (which expects raw input) or the AdminUI (which throws this error https://github.com/vendure-ecommerce/vendure/issues/2473)

**Environment (please complete the following information):**
 - @vendure/core version: 2.1.4
 - Nodejs version v18.11.0
 - Database (mysql/postgres etc): PostgreSQL

**Comment by yasserlens** [comment]
This sounds like a problem that any store with "eager" customEntity relations would face. To solve this it seems we can do one of 2 things:
1) [breaking change] Make sure that state transition events do NOT load/hydrate custom entities of type "relation".
2) [non-breaking] Create a new, lean state transition method which holds the order code and/or order line IDs only. And it's up to the handler to inject services and load data as needed.

@michaelbromley Do you think proposed solution #2 is feasible in the short run?

**Comment by michaelbromley** [comment]
Hi @yasserlens,

Can I just check something first - are your relation custom fields set to be eagerly-loaded (`eager: true`)?

**Comment by yasserlens** [comment]
> Hi @yasserlens,
> 
> Can I just check something first - are your relation custom fields set to be eagerly-loaded (`eager: true`)?

Yes they are (sorry if my last comment wasn't clear - that's what I meant by "eager" customEntity relations). I remember needing the "eager: true" flag early on but I don't remember why.

**Comment by yasserlens** [comment]
tl;dr I solved this by implementing the following middleware in the worker service ONLY:
```typescript
export const config: VendureConfig = {
	apiOptions: {
		...(isRunningInWorker ? {
			middleware: [{
				handler: json({ limit: '10mb' }),
				route: '*',
				beforeListen: true,
			}],
		} : {})
	},
};
```

I used to run both the worker and server instances together on the same service, so I separated them into 2 different services and I added the above which solved the size issue for now (until the server has to deal with >100kb issues).

FYI - The setup I have is on Google Cloud Run, and queues are managed by Google Task Queues. The worker's job is to queue tasks into Google Task Queues.

Also FYI - One more thing I tried which did NOT work: In a plugin, I listened for state change events, and fired a custom event with the order code only. Since the original request (which had large customEntities eagerly loaded) was too large for the worker to process.
